lib/IDE/SyntaxModel.cpp

```diff
@@ -605,6 +602,17 @@ std::pair<bool, Expr *> ModelASTWalker::walkToExprPre(Expr *E) {
           SM, CE->getArgs()->getSourceRange());
     pushStructureNode(SN, CE);
 
+  } else if (auto *SE = dyn_cast<SubscriptExpr>(E)) {
+    SyntaxStructureNode SN;
+    SN.Kind = SyntaxStructureKind::CallExpression;
```

Comment:
Not sure whether we actually want to add this or just ignore it entirely since it isn't actually a `CallExpression`.

Reply:
> Not sure whether we actually want to add this or just ignore it entirely since it isn't actually a `CallExpression`.

Would we want a new syntax structure kind for subscripts, or is the intent for it to denote a "call-like" expression?

> We may also want to add the parent to the `ArgumentInfo` map and check that it's the same when we do the argument lookup, since there's nothing preventing a similar problem on other expressions at the moment.

Sounds good to me!

Reply:
Seems like "call-like" is reasonable. I'll put up another PR to either track the parent or go back to finding the arg like it was before.